packages/mask/src/plugins/RedPacket/SNSAdaptor/RedPacketInjection.tsx

```diff
@@ -0,0 +1,17 @@
+import { useEffect, useState } from 'react'
+import { emitter } from './emitter.js'
+import RedPacketDialog from './RedPacketDialog.js'
+
+export function RedPacketInjection() {
+    const [open, setOpen] = useState(false)
+
+    useEffect(() => {
+        const unsubscribe = emitter.on('open', () => setOpen(true))
+        return () => {
+            unsubscribe()
+        }
```

Comment:
```suggestion
        return emitter.on('open', () => setOpen(true))
```

Reply:
No, it can't `Type '() => boolean' is not assignable to type 'void | Destructor'.`